src/mappings/mappingHandlers.ts

```diff
@@ -6,39 +6,41 @@ import { createExtrinsic } from './entities/mapExtrinsic';
 import mapSubqueryVersion from './entities/mapSubqueryVersion';
 import genesisHandler from './migrations/genesisHandler';
 import { logError } from './util';
-import migrationHandlers from './migrations/migrationHandlers';
 
-let lastBlockId = -1;
+let lastBlockHash = '';
 let lastEventIdx = -1;
+let startupHandled = false;
 
 export async function handleGenesis(): Promise<void> {
   await genesisHandler().catch(e => logError(e));
 }
 
+export async function handleMigration(substrateEvent: SubstrateEvent): Promise<void> {
+  /**
+   * In case of major chain upgrade, we need to process some entities
+   */
+  await mapChainUpgrade(substrateEvent).catch(e => logError(e));
+}
+
+export async function handleStartup(): Promise<void> {
+  /**
+   * This handles the insertion of new SQ version on every restart
+   */
+  await mapSubqueryVersion().catch(e => logError(e));
+}
+
 export async function handleEvent(substrateEvent: SubstrateEvent): Promise<void> {
-  const header = substrateEvent.block.block.header;
-  const blockId = header.number.toNumber();
+  if (!startupHandled) {
+    await handleStartup();
+    startupHandled = true;
```

Comment:
I noticed that, from the time we stopped adding semantic bot commits the version in `package.json` doesn't get updated. So we need to either update that manually or keep back the git commits from bot to make sure the version is updated correctly. 

We use this version as MIN_SQ_VERSION in sdk to check if the SQ is compatible with SDK functions.

Reply:
makes sense. I'll bump it when I am ready to promote all of the alpha optimizations to master